src/sagemaker_xgboost_container/algorithm_mode/callback.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import os
+import tempfile
+import threading
+import queue
+from typing import Callable, Optional
+from xgboost import rabit
+from xgboost.core import CallbackEnv
+from xgboost.callback import _fmt_metric as fmt_metric
+
+
+TEMP_FILE_SUFFIX = ".sagemaker-ignore"
+FILE_LOCK_SUFFIX = ".sagemaker-uploading"
+FILE_SAFE_SUFFIX = ".sagemaker-uploaded"
+
+CHECKPOINT_FILENAME = "xgboost-checkpoint"
+CHECKPOINT_NUM_DIGITS = 12
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def _zero_pad(iteration: int) -> str:
```

Comment:
Method names should ideally start with an action, i.e. '_add_zero_pad' unless it is a property